apps/src/dance/Dance.js

```diff
@@ -451,24 +455,41 @@ Dance.prototype.shouldShowSharing = function () {
  * This is called while this.p5 is in the preload phase.
  */
 Dance.prototype.onP5Preload = function () {
-  const getSelectedSong = () => getStore().getState().selectedSong;
-
   this.nativeAPI = new DanceParty(this.p5, {
-    getSelectedSong,
     onPuzzleComplete: this.onPuzzleComplete.bind(this),
     playSound: audioCommands.playSound,
     recordReplayLog: this.shouldShowSharing(),
   });
+  this.updateSongMetadata(getStore().getState().selectedSong);
   const spriteConfig = new Function('World', this.level.customHelperLibrary);
   this.nativeAPI.init(spriteConfig);
   this.nativeAPI.preload();
 };
 
+Dance.prototype.updateSongMetadata = function (id) {
+  this.songMetadataPromise = this.loadSongMetadata(id);
+};
+
+Dance.prototype.playSong = function (file, callback) {
```

Comment:
`playSong` is not used in this set of changes.

Reply:
good point - I'll remove it.